app/experimenter/kinto/tasks.py

```diff
@@ -78,54 +78,46 @@ def nimbus_check_kinto_push_queue():
     for application, collection in NimbusExperiment.KINTO_APPLICATION_COLLECTION.items():
         kinto_client = KintoClient(collection)
 
-        rejected_collection_data = kinto_client.get_rejected_collection_data()
-        if rejected_collection_data:
-            rejected_slug = kinto_client.get_rejected_record()
-            experiment = NimbusExperiment.objects.get(slug=rejected_slug)
-            if (
-                experiment.status == NimbusExperiment.Status.LIVE
-                and experiment.is_end_requested
-            ):
-                experiment.is_end_requested = False
-            else:
-                experiment.status = NimbusExperiment.Status.DRAFT
-
-            experiment.save()
-
-            generate_nimbus_changelog(
-                experiment,
-                get_kinto_user(),
-                message=f'Rejected: {rejected_collection_data["last_reviewer_comment"]}',
-            )
+        if kinto_client.has_pending_review():
+            continue
+        elif rejected_collection_data := kinto_client.get_rejected_collection_data():
+            handle_rejection.delay(collection, rejected_collection_data)
```

Comment:
In your example you weren't passing the rejected data through to the other task. Is this what you meant? Or did you intend to have kinto make another call in the new task?

Reply:
Oh yeah, I mean we have to instantiate the client and get the rejection data inside the rejection task anyways so I don't think you need to pass it through, since there will always only ever be one pending rejection, so you can just say

```py
elif kinto_client.has_rejection()
```

and invoke the task and then find the rejection data from the collection inside the rejection task.


The other thing is.... In theory, in a single invocation of the parent task we could both clear the rejection out of the collection AND push a new change to it.  Structuring it like this will only clear the rejection out then end, and the next invocation would push something.

So... we might not want to make the rejection handling a separate task in which case it'd just be something like


```py
if kinto_client.has_pending_review():
    continue

if kinto_client.has_rejection():
    handle_rejection(kinto_client)

if queued_launch......
```


In *theory* we could make the rejection handling a separate task, AND do two tasks in a single invocation by using some celery mechanisms to dispatch multiple ordered tasks, but I don't think we need to get that fancy.  In theory we don't need to dispatch any subtasks from this parent task at all, but it's kind of a nice way to isolate things from a metrics/blocking/error handling perspective.

So yeah that's a long winded way of saying, try my example above of rejection handling just as a function call rather than a task, but for the rest of these yes looks great 👏 👏 👏